### PR TITLE
 Add support for aliasing languages like C/C++, Java/Kotlin, and JS/TS

### PR DESCRIPTION
**Pull request by henrymercer**
This PR adds support for the `c-cpp`, `java-kotlin`, and `javascript-typescript` aliases for C/C++, Java/Kotlin, and JavaScript/TypeScript analysis.  To ensure a consistent experience across all CodeQL products, we implemented these aliases in the CLI, so this PR is really just exposing that functionality.

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: src/codeql.ts [hunk 1]
```diff
@@ -330,6 +333,11 @@ export const CODEQL_VERSION_RESOLVE_ENVIRONMENT = "2.13.4";
  */
 export const CODEQL_VERSION_LANGUAGE_BASELINE_CONFIG = "2.14.2";
 
+/**
+ * Versions 2.14.4+ of the CodeQL CLI support language aliasing.
+ */
+export const CODEQL_VERSION_LANGUAGE_ALIASING = "2.14.4";
```
Comment (angelapwen) [inline src/codeql.ts:339]: Hm.. should we bump this to 2.14.5 given its rollout on dotcom was stopped and bumped directly to 2.14.5?

File: src/config-utils.ts [hunk 2]
```diff
@@ -874,7 +870,7 @@ export function getNoLanguagesError(): string {
 }
 
 export function getUnknownLanguagesError(languages: string[]): string {
-  return `Did not recognise the following languages: ${languages.join(", ")}`;
+  return `Did not recognize the following languages: ${languages.join(", ")}`;
```
Comment (angelapwen) [inline src/config-utils.ts:873]: 🇺🇸

File: pr-checks/checks/language-aliases.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+name: "Language aliases"
+description: "Tests that ML-powered queries are run with the security-extended suite and that they produce alerts on a test DB"
```
Comment (angelapwen) [inline pr-checks/checks/language-aliases.yml:2]: I think this description needs to be updated 🤔

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,7 @@ See the [releases page](https://github.com/github/codeql-action/releases) for th
 ## [UNRELEASED]
 
 - Update default CodeQL bundle version to 2.14.5. [#1882](https://github.com/github/codeql-action/pull/1882)
+- Enable the following language aliases when using CodeQL 2.14.4 and later: `c-cpp` for C/C++ analysis, `java-kotlin` for Java/Kotlin analysis, and `javascript-typescript` for JavaScript/TypeScript analysis. [#1883](https://github.com/github/codeql-action/pull/1883)
```
Comment (angelapwen) [inline CHANGELOG.md:8]: Same comment as above on whether we want to move this to 2.14.5.

**Review by angelapwen — DISMISSED** [review]
Approving to unblock, but have a couple of comments. Please feel free to get approval from another 🇪🇺 person once addressed!

File: src/codeql.ts (hunk 1 above)
Comment (henrymercer) [reply src/codeql.ts:339]: 2.14.5 is identical to 2.14.4 apart from a JavaScript fix, so I think it's fine to use 2.14.4 here.  As you say though, it doesn't affect that many customers.

**Review by Z80coder — APPROVED** [review]

**Pull request merged**